pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -172,10 +174,33 @@ func (o *objectWatcherImpl) Update(ctx context.Context, clusterName string, desi
 
 	desireObj, err = o.retainClusterFields(desireObj, clusterObj)
 	if err != nil {
-		klog.Errorf("Failed to retain fields for resource(kind=%s, %s/%s) in cluster %s: %v", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName, err)
+		klog.Errorf("Failed to retain fields for desired resource(kind=%s, %s/%s) in cluster %s: %v", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName, err)
 		return OperationResultNone, err
 	}
 
+	clusterObjCopy := clusterObj.DeepCopy()
+	err = prune.RemoveIrrelevantFields(clusterObjCopy, prune.RemoveJobTTLSeconds)
+	if err != nil {
+		klog.Errorf("Failed to remove irrelevant fields for cluster resource(kind=%s, %s/%s) in cluster %s: %v", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName, err)
+		return OperationResultNone, err
+	}
+
+	// call retain interpret hook again.
+	clusterObjCopy, err = o.retainClusterFields(clusterObjCopy, clusterObj)
+	if err != nil {
+		klog.Errorf("Failed to retain fields for cluster resource(kind=%s, %s/%s) in cluster %s: %v", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName, err)
+		return OperationResultNone, err
+	}
```

Comment:
This (call retain interpret hook) is concerning, as it is a remote call if the interpreter is implemented as a webhook, and in such as a case, it's an expensive call.

Reply:
This is equivalent to replacing the request to the member cluster apiserver with a request to interpretwebhook, since the request to interpretwebhook does not occupy the rate limit, there should still be some benefits.